raytracing/matrix.py

```diff
@@ -378,6 +380,62 @@ def _waitForFile(self, filePath, timeout=5):
             oldSize = currentSize
         time.sleep(0.1)
 
+    def traceManyThroughInMultiprocess(self, inputRays, processes=4):
+        """
+        Naive implementation of multiprocessing on `inputRays`.
+        """
+        with multiprocessing.Pool(processes=processes) as pool:
+            outputRays = pool.map(self.traceThrough, inputRays)
```

Comment:
This here does not work because different processes access inputRays at once, and each access calls __iter__, which modifies `self.iteration`.  Sometimes, the clash, and the computation never stops.  This happens to me frequently.  This is because RandomRays() is not thread-safe: we must lock access to the __iter__ function OR expand the list explicitly.  

I looked at using Lock() but we can't do it simply because the lock cannot be used across processes (they use a file to exchange information, like I did originally but better). Therefore, I simply expanded the list explicitly to keep the code simple.